sdk/rules_daml/daml.bzl

```diff
@@ -54,8 +56,7 @@ build-options: [{opts}]
             dependencies = ", ".join(dependencies),
             data_dependencies = ", ".join(data_dependencies),
             module_prefixes = "\n".join(["  {}: {}".format(k, v) for k, v in module_prefixes.items()]),
-            upgrades = "upgrades: " + upgrades if upgrades else "",
-            typecheck_upgrades = "typecheck-upgrades: false" if not typecheck_upgrades else "",
+            upgrades = "upgrades: " + upgrades if upgrades and using_local_compiler(target) else "",
```

Comment:
Does the local compiler not run upgrades checks? I see you're not allowing either the upgrades field or the typecheck-upgrades field when in compile only.

Reply:
The local compiler is legacy 2.8.3, so it doesn't recognize the upgrades-typecheck flag since we just added it. Back when this was a daml.yaml option, it would be ignored, but now that it's a flag, it breaks the compiler.